server/src/main/java/keywhiz/service/daos/SecretDAO.java

```diff
@@ -419,6 +472,14 @@ public ImmutableList<SecretSeriesAndContent> getSecrets(@Nullable Long expireMax
     });
   }
 
+  public ImmutableList<SecretSeriesAndContent> getSecrets(@Nullable Long expireMaxTime,
+      @Nullable Group group, @Nullable Long expireMinTime, @Nullable String minName,
+      @Nullable Integer limit, Object principal) {
+    permissionCheck.checkAllowedOrThrow(principal, Action.READ, null);
+
+    return getSecrets(expireMaxTime, group, expireMinTime, minName, limit);
+  }
```

Comment:
Why is it not good to return ImmutableList?

Reply:
For collections, you generally want to always return the top-level interface (List) rather than the specific implementation so that your callers don't come to rely on some specific behavior of that implementation. This increases flexibility because you can then swap out to whatever implementation you'd like and your callers should be unaffected.